zap/src/main/java/org/zaproxy/zap/spider/parser/SpiderHtmlParser.java

```diff
@@ -208,13 +239,44 @@ private boolean parseSource(HttpMessage message, Source source, int depth, Strin
             resourcesfound |= processAttributeElement(message, depth, baseURL, el, "src");
         }
 
+        // Process Table elements
+        elements = source.getAllElements(HTMLElementName.TABLE);
+        for (Element el : elements) {
+            resourcesfound |= processAttributeElement(message, depth, baseURL, el, "background");
+            Source tableSource = new Source(el.getContent().toString());
```

Comment:
Why not just use do `source.getAllElements(HTMLElementName.TD)` ?
We dont care is a 'TD' element is not in a table, we just want to know if it has a `background` element 😄

Reply:
(at the same level ypou process the 'TABLE' elements of course ;)

Reply:
Because I was not sure what we were aiming at and I was trying to be as compliant as possible with the HTML spec :) 
If I can just search for TD elements, so be it